jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageWriter.java

```diff
@@ -75,21 +77,36 @@ private WrittenLayer(
    * @throws IOException if an I/O exception occurs
    */
   private static void moveIfDoesNotExist(Path source, Path destination) throws IOException {
-    // If the file already exists, we skip renaming and use the existing file. This happens if a
-    // new layer happens to have the same content as a previously-cached layer.
-    if (Files.exists(destination)) {
+    // Some Windows users report java.nio.file.AccessDeniedException that we suspect is caused
+    // by anti-virus programs, like Windows Defender, that open new files for scanning.
+    // Retry the rename up to 5 times, with 15ms pause between each retry.
+    boolean success =
+        Retry.action(
+                () -> {
+                  if (Files.exists(destination)) {
+                    // If the file already exists, we skip renaming and use the existing file.
+                    // This happens if a new layer happens to have the same content as a
+                    // previously-cached layer.
+                    return true;
+                  }
+                  Files.move(source, destination);
+                  return Files.exists(destination);
+                })
+            .maximumRetries(5)
+            .retryOnException(ex -> ex instanceof FileSystemException)
+            .sleep(15, TimeUnit.MILLISECONDS)
+            .run();
+    if (success) {
       return;
     }
 
-    try {
-      Files.move(source, destination);
-
-    } catch (FileSystemException ex) {
-      if (!Files.exists(destination)) {
-        // TODO to log that the destination exists
-        throw ex;
-      }
-    }
+    String message =
+        String.format(
+            "unable to move: %s to %s; this is often caused by interference from antivirus",
+            source, destination);
+    // TODO figure out a better logging scheme
+    Logger.getLogger(CacheStorageWriter.class.getName()).warning(message);
```

Comment:
Ugh... I didn't know this class does not accept `EventHandlers`. Does this make Maven/Gradle print out a warning? Otherwise, I think this is not so meaningful. And I feel a bit awkward because we don't use `Logger.getLogger` anywhere else in our codebase.

In any case, I was hoping that we could print out the warning whenever there is a retry (not that when all the retries fail). Maybe we can discuss this in more detail later.

Reply:
I'll remove this.  The `IOException` should be propagated upwards with the warning.

Reply:
If I force the move action to fail, Jib reports the right thing:

> [ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.4.1-SNAPSHOT:buildTar (default-cli) on project first: unable to move: /Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/maven/app/target/jib-cache/tmp/4013610408160898804/.tmp.layer to /Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/maven/app/target/jib-cache/tmp/4013610408160898804/932da51564135c98a49a34a193d6cd363d8fa4184d957fde16c9d8527b3f3b02; this is often caused by interference from antivirus

Reply:
Tweaked the message

Reply:
does this still need to be removed?